docs/api/ethereum.md

```diff
@@ -1,5 +1,645 @@
+
+A static class `function` that returns a new instance of the BitskiProvider class (alias for `new BitskiProvider`).
+
+**Arguments**
+
+See the class [constructor](#bitski-provider) for details.
+
+**Usage**
+
+```ts
+import { BitskiProvider } from '@0xcert/ethereum-bitski-backend-provider';
+
+// create provider instance
+const provider = BitskiProvider.getInstance();
+```
+
+### getNetworkVersion()
+
+An `asynchronous` class instance `function` which returns Ethereum network version (e.g. `1` for Ethereum Mainnet).
+
+**Result:**
+
+A `string` representing Ethereum network version.
+
+**Example:**
+
+```ts
+// perform query
+const version = await provider.getNetworkVersion();
+```
+
+### isCurrentAccount(accountId)
+
+A `synchronous` class instance `function` which returns `true` when the provided `accountId` maches the currently set account ID.
+
+**Arguments:**
+
+| Argument | Description
+|-|-
+| accountId | [required] A `string` representing Ethereum account address.
+
+**Result:**
+
+A `boolean` which tells if the `accountId` maches the currently set account ID.
+
+**Example:**
+
+```ts
+// ethereum wallet address
+const walletId = '0x06012c8cf97bead5deae237070f9587f8e7a266d';
+
+// perform query
+const maches = provider.isCurrentAccount(walletId);
+```
+
+### isSupported()
+
+A `synchronous` class instance `function` which returns `true` when the provider is supported by the environment.
+
+**Result:**
+
+A `boolean` which tells if the provider can be used.
+
+**Example:**
+
+```ts
+// perform query
+const isSupported = provider.isSupported();
+```
+
+### isUnsafeRecipientId(ledgerId)
+
+A `synchronous` class instance `function` which returns `true` when the provided `id` is listed among unsafe recipient ids on the provided.
+
+**Arguments:**
+
+| Argument | Description
+|-|-
+| ledgerId | [required] A `string` representing an address of the ERC-721 related smart contract on the Ethereum blockchain.
+
+**Result:**
+
+A `boolean` which tells if the `id` is unsafe recipient.
+
+**Example:**
+
+```ts
+// unsafe recipient address
+const criptoKittiesId = '0x06012c8cf97bead5deae237070f9587f8e7a266d';
+
+// perform query
+const isUnsafe = provider.isUnsafeRecipientId(criptoKittiesId);
+```
+
+**See also:**
+
+[unsafeRecipientIds](#unsaferecipientids)
+
+### mutationTimeout
+
+A class instance `variable` holding an `integer` number of milliseconds in which a mutation times out.
+
+### on(event, handler);
+
+A `synchronous` class instance `function` which attaches a new event handler.
+
+**Arguments:**
+
+| Argument | Description
+|-|-
+| event | [required] A `string` representing a [provider event](./ethereum.md#provider-events) name.
+| handler | [required] A callback `function` which is triggered on each `event`. When the `event` equals `ACCOUNT_CHANGE`, the first argument is a new account ID and the second is the old one, when the `event` equals `NETWORK_CHANGE`, the first argument is a new network version and the second is the old one.
```

Comment:
`| handler | [required] A callback `function` which is triggered on each `event`. When the `event` equals `ACCOUNT_CHANGE`, the first argument is a new account ID and the second argument is the old ID, and when the `event` equals `NETWORK_CHANGE`, the first argument is a new network version and the second argument is the old version.`